pip/req/req_set.py

```diff
@@ -447,6 +474,12 @@ def _prepare_file(self, finder, req_to_install):
                 if self.is_download:
                     req_to_install.archive(self.download_dir)
             elif req_to_install.satisfied_by:
+                if require_hashes:
+                    logger.info(
+                        'Since it is already installed, we are trusting this '
+                        'package without checking its hash. To ensure a '
+                        'completely repeatable environment, install into an '
+                        'empty virtualenv.')
```

Comment:
So this doesn't actually detect an empty virtual environment, it just detects one where the thing you're installing happens to already exist in the virtual environment. There could be any number of things installed (and for that matter, the idea of an actually empty virtual environment is a bit of a strange duck, since presumably you have at least pip and setuptools already installed, if not wheel too.

I think it's way too much for ``INFO`` on every requirement, I could possibly live with a single ``INFO`` message, but think it should probably be a ``DEBUG`` most likely.